src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getFruit() == null) {
+            throw new RuntimeException("There are no "
+                    + fruitTransaction.getFruit() + " int the storage");

```

Comment:
There's a logical issue here. If `fruitTransaction.getFruit()` returns null, concatenating this null with a string in the RuntimeException message will result in the message 'There are no null int the storage', which is not informative. Instead, you should check for null before attempting to get the fruit from the storage, and throw an exception with a message like 'FruitTransaction contains no fruit.'